model/testing/src/icon4py/model/testing/serialbox.py

```diff
@@ -69,8 +69,12 @@ def wrapper(self, *args, **kwargs):
                         f"{name}: field not registered in savepoint {self.savepoint.metainfo}"
                     )
                     if dims:
-                        shp = tuple(self.sizes[d] for d in dims)
-                        return gtx.as_field(dims, np.zeros(shp), allocator=self.backend)
+                        # We allocate a dummy field with size 1 in each dimension
```

Comment:
what do you think about this?

Reply:
Yes, I checked that if the user disables the code-path that uses these fields, they are not accessed in the generated code.

Longer term we want to have optional fields, but we didn't think of an interface. In the example of this PR, the field would be only optional if `apply_zdiffusion_t==False`. So maybe we want syntax like `OptionalIf[Field[...], condition]`. Then, should the condition be required to be compile-time etc...